java/src/main/java/org/rocksdb/RocksDB.java

```diff
@@ -324,7 +325,19 @@ public static RocksDB open(final DBOptions options, final String path,
     }
 
     db.ownedColumnFamilyHandles.addAll(columnFamilyHandles);
-    db.storeDefaultColumnFamilyHandle(db.makeDefaultColumnFamilyHandle());
+
+    // ColumnFamilyHandle.isDefaultColumnFamily() doesn't work here yet, as we are in process of
+    // opening database
+    OptionalInt defaultCfIndex = IntStream.of(0, columnFamilyDescriptors.size() - 1)
+                                     .filter(x
+                                         -> Arrays.equals(columnFamilyDescriptors.get(x).getName(),
+                                             RocksDB.DEFAULT_COLUMN_FAMILY))
+                                     .findFirst();
+    if (defaultCfIndex.isPresent()) {
+      db.storeDefaultColumnFamilyHandle(columnFamilyHandles.get(defaultCfIndex.getAsInt()));
+    } else {
+      throw new RocksDBException("No default column family");
```

Comment:
message -> "You must provide the default column family in your columnFamilyDescriptors"

Reply:
Let's move the check for the default column family before the `open` in the loop for the descriptors